bndtools.core/src/org/bndtools/core/templating/repobased/ReposTemplateLoader.java

```diff
@@ -172,23 +173,30 @@ private static void addPreferenceConfiguredRepos(List<Repository> repos, Reporte
         if (bndPrefs != null && bndPrefs.getEnableTemplateRepo()) {
             List<String> repoUris = bndPrefs.getTemplateRepoUriList();
             try {
-                FixedIndexedRepo prefsRepo = loadRepo(repoUris);
+                OSGiRepository prefsRepo = loadRepo(repoUris, reporter);
                 repos.add(prefsRepo);
-            } catch (IOException | URISyntaxException ex) {
+            } catch (Exception ex) {
                 reporter.exception(ex, "Error loading preference repository: %s", repoUris);
             }
         }
     }
 
-    private static FixedIndexedRepo loadRepo(List<String> uris) throws IOException, URISyntaxException {
-        FixedIndexedRepo repo = new FixedIndexedRepo();
+    private OSGiRepository loadRepo(List<String> uris, Reporter reporter) throws Exception {
+        OSGiRepository repo = new OSGiRepository();
+        repo.setReporter(reporter);
+        repo.setRegistry(workspace);
+
         StringBuilder sb = new StringBuilder();
         for (Iterator<String> iter = uris.iterator(); iter.hasNext();) {
             sb.append(iter.next());
             if (iter.hasNext())
                 sb.append(',');
         }
-        repo.setLocations(sb.toString());
+
+        Map<String,String> properties = new HashMap<>();
+        properties.put("name", "bndtools-preferences-template-repos");
+        properties.put("locations", sb.toString());
```

Comment:
I wasn't completely sure here .. might be safer to also add `max.stale=0` to prevent caching as caching for this repo can't be configured?